### PR TITLE
inject lost prev file information into the current file snapshot

### PR DESCRIPTION
**Pull request by fvacek**
value is set in previous file, but it is not present in current file
snapshot this can happen if device was switched off and for example error
set to true was cleared meanwhile
we have to inject this lost information into the current file snapshot

### CODE REVIEW

File: libshvcore/src/utils/shvfilejournal.cpp [hunk 1]
```diff
@@ -757,27 +762,50 @@ chainpack::RpcValue ShvFileJournal::getLog(const ShvFileJournal::JournalContext
...
+				else if(!prev_file_values.empty()) {
+					for(const auto &kv : prev_file_values) {
+						ShvJournalEntry e3 = kv.second;
+						if(!e3.value.hasDefaultValue() && this_file_values.find(e3.path) == this_file_values.cend()) {
+							// value is set in previous file, but it is not present in current file snapshot
+							// this can happen if device was switched off and for example error set to true
+							// was cleared meanwhile
+							// we have to inject this lost information into the current file snapshot
+							e3.value.setDefaultValue();
+							entries_to_write.push_back(kv.second);
```
Comment (MartinKT) [inline libshvcore/src/utils/shvfilejournal.cpp:780]: Myslim ze tady se do e3 musi nastavit cas m_snapShotEpochMsec. Pro funkci `write_snapshot` by to nevadilo, ale mohlo by to vadit pri porovnavani na radce 786 a ve funkci `append_log_entry`.

File: libshvcore/src/utils/shvfilejournal.cpp [hunk 2]
```diff
@@ -757,27 +762,50 @@ chainpack::RpcValue ShvFileJournal::getLog(const ShvFileJournal::JournalContext
...
+				this_file_values[e.path] = e;
+				entries_to_write[0] = e;
+				entries_to_write.resize(1);
+				if(rd.isInSnapShot()) {
+					prev_file_values.erase(e.path);
+				}
+				else if(!prev_file_values.empty()) {
+					for(const auto &kv : prev_file_values) {
+						ShvJournalEntry e3 = kv.second;
+						if(!e3.value.hasDefaultValue() && this_file_values.find(e3.path) == this_file_values.cend()) {
```
Comment (MartinKT) [inline libshvcore/src/utils/shvfilejournal.cpp:774]: Ta druha cast podminky je zbytecna, ne? Pokud se to dostalo do `this_file_values` tak uz je to vymazane z `prev_file_values`? Nechci byt puntickar, pokud to tam mas pro jistotu, tak mi to nevadi, jen jestli to spravne chapu.

File: libshvcore/src/utils/shvfilejournal.cpp (hunk 1 above)
Comment (fvacek) [reply libshvcore/src/utils/shvfilejournal.cpp:812]: spravna pripominka :+1:

File: libshvcore/src/utils/shvfilejournal.cpp (hunk 2 above)
Comment (fvacek) [reply libshvcore/src/utils/shvfilejournal.cpp:806]: jj

File: libshvcore/src/utils/shvfilejournal.cpp (hunk 1 above)
Comment (fvacek) [reply libshvcore/src/utils/shvfilejournal.cpp:815]: opraveno

File: libshvcore/src/utils/shvfilejournal.cpp (hunk 2 above)
Comment (fvacek) [reply libshvcore/src/utils/shvfilejournal.cpp]: opraveno 3393e66

**Pull request closed without merging**